### PR TITLE
feat(kinesisfirehose): add metrics functions to IDeliveryStream

### LINKED ISSUE #15543: (aws-kinesisfirehose): support for metrics in DeliveryStream
### Use Case

Implement specific metrics functions: i.e. `metricIncomingBytes()`

### Proposed Solution

See: https://github.com/aws/aws-cdk/pull/15505/files

* [ ] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by madeline-k**
Closes #15543 

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-kinesisfirehose/README.md [hunk 1]
```diff
@@ -152,6 +152,43 @@ new DeliveryStream(this, 'Delivery Stream', {
...
+import * as cloudwatch from '@aws-cdk/aws-cloudwatch';
+// Alarm that triggers when the per-second average of incoming bytes exceeds 90% of the current service limit
+const incomingBytesPercentOfLimit = new cloudwatch.MathExpression({
+  expression: 'incomingBytes / 300 / bytePerSecLimit',
+  usingMetrics: {
+    incomingBytes: deliveryStream.metricIncomingBytes({ statistic: cloudwatch.Statistic.SUM }),
+    bytePerSecLimit: deliveryStream.metric('BytesPerSecondLimit'),
+  },
+});
+new Alarm(this, 'Alarm', {
```
Comment (BenChaimberg) [inline packages/@aws-cdk/aws-kinesisfirehose/README.md:182]:
```suggestion
new cloudwatch.Alarm(this, 'Alarm', {
```

File: packages/@aws-cdk/aws-kinesisfirehose/test/delivery-stream.test.ts [hunk 2]
```diff
@@ -178,6 +179,100 @@ describe('delivery stream', () => {
         },
       });
     });
+    test('metricIncomingBytes', () => {
```
Comment (skinny85) [inline packages/@aws-cdk/aws-kinesisfirehose/test/delivery-stream.test.ts:182]: Empty line before this line.

File: packages/@aws-cdk/aws-kinesisfirehose/test/delivery-stream.test.ts [hunk 3]
```diff
@@ -178,6 +179,100 @@ describe('delivery stream', () => {
...
+        dimensions: {
+          DeliveryStreamName: deliveryStream.deliveryStreamName,
+        },
+      });
+    });
+
+    test('metricBackupToS3Records', () => {
+      const deliveryStream = new firehose.DeliveryStream(stack, 'Delivery Stream', {
+        destinations: [mockS3Destination],
+      });
```
Comment (skinny85) [inline packages/@aws-cdk/aws-kinesisfirehose/test/delivery-stream.test.ts:261]: These 3 lines are repeated in every single test.

How about adding a `describe()` here, and then a `beforeEach()` in it to get rid of this duplication?

**Review by skinny85 — APPROVED** [review]
Looks fantastic! 3 minor comments.

**Review by skinny85 — APPROVED** [review]

**Pull request merged**